src/weapon/w_020.c

```diff
@@ -3,12 +3,422 @@
 #include "weapon_private.h"
 #include "shared.h"
 
+extern SpriteParts D_90000_8017A040[];
+extern AnimationFrame D_90000_8017A850[];
+extern AnimationFrame D_90000_8017A864[];
+extern AnimationFrame D_90000_8017A8C8[];
+extern AnimationFrame D_90000_8017A8D8[];
+extern AnimationFrame D_90000_8017A8E8[];
 extern AnimationFrame D_90000_8017A954[];
 extern AnimationFrame D_90000_8017A9D8[];
+extern FrameProperty D_90000_8017A9E0;
 extern u16 D_90000_8017AA10[];
+extern u8 D_90000_8017AA1C;
+extern s32 D_90000_8017AADC[];
+extern s32 D_90000_8017AB44[];
 extern s32 D_90000_8017C238;
 
-INCLUDE_ASM("weapon/nonmatchings/w_020", EntityWeaponAttack);
+void EntityWeaponAttack(Entity* self) {
+    Collider sp10;
+    Entity* child;
+    Primitive* prim;
+
+    u16 var_s4;
+    s16 angle_s3;
+    s32 var_s2;
+    u8* var_s1;
+
+    s16 xVar;
+    s16 yVar;
+
+    s16 xOffset;
+    s16 yOffset;
+    s16 sp6a;
+    s16 sp68;
+    s16 sp66;
+    s16 sp64;
+    s16 sp62;
+    u16 sp60;
+    s16 var_s8;
+    s16 var_s7;
+
+    sp60 = D_90000_8017AA10[g_GameTimer % 3] & 0x7FFF;
```

Comment:
i feel like modulus makes more sense in these 0x7FFF i've been seeing around, idk what you think.
```suggestion
    sp60 = D_90000_8017AA10[g_GameTimer % 3] % 0x8000;

Reply:
Sure, why not, I'll make the change.

Reply:
Although it seems it doesn't work in cases like the rand() below, that's a pitty.

Reply:
What's the rationale for using modulus? This gets used like:
```
prim->clut = sp60;
```
The top bit of the clut is supposed to be 0 so it seems like it's getting masked off

https://psx-spx.consoledev.net/graphicsprocessingunitgpu/#clut-attribute-color-lookup-table-aka-palette